src/test_application/main.cpp

```diff
@@ -291,6 +310,37 @@ void debug_print_update_order()
     }
 }
 
+void debug_print_machines()
+{
+    using osp::active::ACompHierarchy;
+    using osp::active::ActiveScene;
+    using osp::active::ActiveEnt;
+    using osp::active::ACompMachines;
+
+    if (!g_ospMagnum)
+    {
+        std::cout << "Can't do that yet, start the magnum application first!\n";
+        return;
+    }
+
+    ActiveScene& scene = g_ospMagnum->get_scenes().begin()->second;
+    auto view = scene.get_registry().view<osp::active::ACompMachines>();
+
+    for (ActiveEnt ent : view)
+    {
+        ACompHierarchy& hier = scene.reg_get<ACompHierarchy>(ent);
+        std::cout << "[" << int(ent) << "]: " << hier.m_name << "\n";
+
+        auto& machines = scene.reg_get<ACompMachines>(ent);
+        for (auto mach : machines.m_machines)
```

Comment:
don't get too crazy with the use of auto, it makes it hard for readers to pick up, via context clues, what's going on.